src/lib/mina/transaction.ts

```diff
@@ -276,6 +280,40 @@ type RejectedTransaction = Pick<
   errors: string[];
 };
 
+type TransactionPromise = Promise<Transaction> & {
```

Comment:
quick question: just wanted to check, are these new types exposed to the user or would it be helpful to do so? or would it be better to just keep this internal?

Reply:
Good thinking! They should indeed be exposed to the user. Just added their reexports.